renku/api/secrets.py

```diff
@@ -0,0 +1,38 @@
+import base64
+import os
+
+import attr
+
+
+@attr.s
+class SecretsApiMixin:
+    """Client for handling secrets for external providers."""
+
+    SECRETS_DIR = 'providers'
+
+    @property
+    def renku_secrets_path(self):
+        return self.renku_path.joinpath(self.SECRETS_DIR)
```

Comment:
do you want this in the repository?

Reply:
No, planned to add them to gitignore.

Reply:
still, there's a reasonable risk that someone will check in their password

Reply:
I know. That's why it's encoded, so it's not super obvious.

If you have a better suggestion I'm all ears. :smile:

Reply:
credentials are per provider, not per dataset or repo - so having them in a renku configuration outside the repo would maybe make more sense?